src/TheThingsNetwork.cpp

```diff
@@ -219,13 +219,14 @@ bool TheThingsNetwork::personalize(const byte devAddr[4], const byte nwkSKey[16]
   return true;
 }
 
-bool TheThingsNetwork::join(const byte appEui[8], const byte appKey[16], long int nbr_delay, int max_attempts) {
+bool TheThingsNetwork::join(const byte appEui[8], const byte appKey[16], bool restart, long int nbr_delay, int max_attempts) {
```

Comment:
We shouldn't add this and just do the same as in #42. Add a proto without the first two args that does no reset and in this one do reset and set the appeui and appkey